### PR TITLE
Added leader

### PR DESCRIPTION
**Pull request by aoaaceai**

### CODE REVIEW

**Comment by secminhr** [comment]
build failed

**Comment by aoaaceai** [comment]
succeeded

File: src/main/java/eod/Game.java [hunk 1]
```diff
@@ -121,6 +123,10 @@ public Player getRivalPlayer(Player player) {
         }
     }
 
+    public boolean isPlayerA(Player unknown) {
+        return unknown == A;
```
Comment (secminhr) [inline src/main/java/eod/Game.java]: Use .equals

File: src/main/java/eod/Gameboard.java [hunk 2]
```diff
@@ -98,13 +103,37 @@ public void removeObject(int x, int y) throws IllegalArgumentException {
...
+    public ArrayList<Point> getSurroundingEmpty(Point p, int range) {
+        ArrayList<Point> points = new ArrayList<>();
+        for(int x = p.x - range;x <= p.x + range;x++) {
+            if(x < 0 || x >= boardSize) {
+                continue;
+            }
+            for(int y = p.y-range; y <= p.y+range;y++) {
+                if(y < 0 || y >= boardSize) {
+                    continue;
+                }
```
Comment (secminhr) [inline src/main/java/eod/Gameboard.java:115]: add empty check

File: src/main/java/eod/Gameboard.java [hunk 3]
```diff
@@ -98,13 +103,37 @@ public void removeObject(int x, int y) throws IllegalArgumentException {
...
+        return points;
+    }
+
+    public void summonObject(WarObject object) throws IllegalArgumentException {
+        int x = object.position.x, y = object.position.y;
+        if(hasObjectOn(x, y)) {
+            throw new IllegalArgumentException("There's already an object on ("+x+", "+y+").");
+        }
+        board[x][y] = object;
+    }
```
Comment (secminhr) [inline src/main/java/eod/Gameboard.java]: consider using x and y as parameter, and assign the position in this method

File: src/main/java/eod/warObject/WarObject.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+//WarObject represented anything on the gameboard
+public abstract class WarObject implements GameObject {
+    public Point position;
+    protected Player player;
+
+    public WarObject(Player player, Point point) {
+        this.player = player;
+        position = point;
+    }
```
Comment (secminhr) [inline src/main/java/eod/warObject/WarObject.java]: WarObject should be created before putting on the board, however, it's impossible to know the position of the object before it is put on the board.
I would suggest removing position as a parameter.

File: src/main/java/eod/Player.java [hunk 5]
```diff
@@ -77,6 +88,21 @@ public boolean isLeaderAlive() {
         return leader.isAlive();
     }
 
+    public ArrayList<Point> getBase() {
```
Comment (secminhr) [inline src/main/java/eod/Player.java]: Suggest rename

**Review by secminhr — APPROVED** [review]
That's a great amount of work.
Good job.

**Pull request merged**